torch/lib/THNN/generic/SpatialClassNLLCriterion.c

```diff
@@ -84,15 +129,45 @@ void THNN_(SpatialClassNLLCriterion_updateGradInput)(
           THNNState *state,
           THTensor *input,
           THIndexTensor *target,
+          THTensor *gradOutput,
           THTensor *gradInput,
           bool sizeAverage,
           THTensor *weights,
           THTensor *total_weight,
-          int64_t ignore_index)
+          int64_t ignore_index,
```

Comment:
It seems that we don't include `TH_INDEX_BASE` in the offset here

Reply:
TH_INDEX_BASE = 0. I think it's used because everything in lua torch is one-indexed? Should I remove TH_INDEX_BASE entirely from this file, or assume that it could potentially equal 1?